lib/cisco_node_utils/portchannel_global.rb

```diff
@@ -0,0 +1,303 @@
+# December 2015, Sai Chintalapudi
+#
+# Copyright (c) 2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative 'node_util'
+
+module Cisco
+  # node_utils class for portchannel_global
+  class PortChannelGlobal < NodeUtil
+    attr_reader :name
+
+    def initialize(name, instantiate=true)
+      fail TypeError unless name.is_a?(String)
+      fail ArgumentError unless name == 'default'
+      @name = name.downcase
+
+      create if instantiate
+    end
+
+    def self.globals
+      hash = {}
+      # if the key already exists, no need create instance
+      hash['default'] =
+          PortChannelGlobal.new('default', false) unless hash.key?('default')
```

Comment:
I don't quite follow this. You just initialized hash in the previous step so it won't have any keys yet.